### DISCUSSION TITLE
[Kavita] Allow multiple book formats

### DISCUSSION THREAD

**Discussion by BrainWart** [Kareadita/Kavita#2631] [Ideas]
### Idea Description

The idea is for a book to be stored as the same item with multiple formats. Download would bring up a drop down to pick the format would you like.

I have multiple books in different formats through Humble Bundle, etc. and this would allow me to keep different versions for different tasks. I could also see this as a way to keep a high resolution copy of a book and a lower resolution/image free copy as well.

### Duration of Using Kavita

< v7.6

### Idea Category

Feature Enhancement

### Agreement

- [X] I agree that this is solely for submitting ideas, and I will search for existing ideas before posting.

**Comment by scare376** [comment]
How would kavita determine what version of the book you want to read? I believe currently how most people handle this is to create different libraries for the different filetypes.

**Reply by BrainWart** [reply]
Honestly, I was just thinking Kavita would pick the "best" format.

I guess one could have a preference option in the reader and leave it up to the "clients" to decide which the user should see. If it was left up to the client, the client on a phone could pick whichever is lightest, and a reader on another device could pick a higher resolution option based on whichever the user had last picked. If looked at from the manga side of things, being able to select a lighter version while roaming would be nice. I use Kavita for books, so I was hoping to have it prefer epub over pdf.

**Reply by DieselTech** [reply]
> Honestly, I was just thinking Kavita would pick the "best" format.

The best format is epub. Too many other formats are dead and unsupported, or are vendor locked down and just plain incompatible. Using the web reader it wouldn't really matter since it will just display on the device anyways.

**Reply by CipherFrame** [reply]
> How would kavita determine what version of the book you want to read? I believe currently how most people handle this is to create different libraries for the different filetypes.

Handle it the same as Plex does with multiple versions; 
![image](https://github.com/user-attachments/assets/dd3ab9c4-d94b-4aab-9911-d381e1cd9337)
Just a simple popup that lets the user click PDF, epub, MOBI etc.

**Comment by GregWoods** [comment]
i must admit I was very surprised this feature didn't exist. Although epub is often the best format, sometimes for technical books, the PDF equivalent is easier to read, so i will often have both formats downloaded from my purchased humble bundle deals

**Reply by scare376** [reply]
If a source/format is better than another. Why keep/serve both? It makes sense to keep different quality levels for plex because of streaming video and you may want to cut down on transcoding.

**Comment by CipherFrame** [comment]
Different use cases, scenarios or end user devices as someone else here
described.

**Comment by majora2007** [comment]
It's not likely I'll make any change to the way Kavita functions. If I do, it would be some time in the future where I see a really valid need for it. Don't take this as me shutting down the idea, please do continue leaving ideas here, but do note that it is not an issue in my eyes currently.

**Comment by scarpentier** [comment]
This is the 3rd most requested feature in the discussions. Publishers often release books in different formats and it is my perspective that this request is not a edge case at all and should be part of Kavita's standard features.

Please reconsider implementing.

**Reply by amodig** [reply]
Indeed, e.g. Calibre supports it. Given that Komga's maintainer has shot down the same feature idea (https://github.com/gotson/komga/issues/497), this could give a certain edge to Kavita.

**Comment by docquantum** [comment]
I would like to add that my most common use case for having multiple formats is that the EPUB doesn't always have the same page numbering nor format as the PDF.

So, especially with scans of the physical book, or where the PDF lines up with the physical book, it's useful to have that reference when discussing the book with others or seeing how formatting might look in some cases, while then using EPUB primarily as it is superior in most cases for reading the book (especially on smaller devices).

I also try to keep around PDF copies as it's more accessible when sharing the file itself not through Kavita, so I try to organize my content such that every book has the AudioBook, PDF, and EPUB in the same folder (or subset if said format doesn't exist).

For what it's worth, AudioBookShelf supports this, but Kavita has a much superior web e-reader setup which is why I would like to use it (plus having Hardcover integration on the horizon is nice).
